Feat/filtering

phoenixAja opened the pull request:
### Description

Self serve tree building and interactive sample filtering

#### Issue
[ch<fill_in_issue_number>](https://app.clubhouse.io/genepi/story/<fill_in_issue_number>)

### Test plan

rdev: https://featfiltering-frontend.dev.genepi.czi.technology

mmmmmmaya committed:
add the most basic filter panel possible to the sample table (#629)

mmmmmmaya committed:
[ch150281] filter panel toggle (#633)

* add filter toggle icon

* create viewname variable enum

* whitespace

* add ability to toggle filter panel

* filter toggle styling

* generalize iconbutton hover style

* remove extraneous types

Commit:
tree build button/ modal / api request (#638)

Commit:
disable tree build button when greater than 2000 samples are selected (#641)

mmmmmmaya committed:
[ch150283] date picker (#642)

* filtering works :)

* types

* remove filter panel placeholder text

* typo

* not useful comment

mmmmmmaya committed:
distribute whitespace around samples table according to design (#646)

mmmmmmaya committed:
[ch152574] add upload filter (#645)

* add upload filter

* fix prop names

mmmmmmaya committed:
[ch150285] lineage filter (#648)

* use new version of czifui in order to pull in menuselect

* new lineage filter component

* add multi filter type to filter panel; add lineage filter logic

* determine lineage list from data

* filter by lineage

* types

* move component to proper location

Commit:
Phoenix/create tree modal (#643)

mmmmmmaya committed:
[ch150281] genome recovery filter (#651)

* bump sds pkg version

* add genome recovery filter

* use transform for date filters

* fix import

* types

mmmmmmaya committed:
Maya/filter styles (#653)

* use standard input dropdowns for all filters

* chip spacing

* optionally hide button if date fields have nothing in them

* date range

* types and lint

mmmmmmaya committed:
Maya/filter styles (#655)

Commit:
fix sample count button (#659)

Commit:
info toast and tree modal updates (#660)

Commit:
👹 fix guide on tree type link 👹 (#666)

vincent-czi committed:
Feat/filtering ux cleanup (#665)

* WIP: Filter UX clean-up

* Fix-ups for linting and TS

* Capitalization tweak for dates

Co-authored-by: Phoenix Logan <phoenix@phoenixlogan.net>

Co-authored-by: maya <mlila@chanzuckerberg.com>
Co-authored-by: Phoenix Logan <phoenix@phoenixlogan.net>

mmmmmmaya committed:
[ch160837] filter state panel toggle (#669)

* dont remove filter state if filter panel hidden

* also don't remove chip state

* types

mmmmmmaya committed:
[ch160824] exclude na dates (#668)

* correct filter order in side panel

* tmp

* exclude na dates from sample filters

* but dont remove them if there isnt any date filter applied

Commit:
reset state on create tree modal close and on mutation fail/success (#671)

Commit:
adjust copy for v1 self serve tree info toast (#672)

vincent-czi committed:
Fix selections in multi-select for lineage filtering  (#670)

Co-authored-by: Timmy Huang <tihuan@users.noreply.github.com>

Janeece commented:
@phoenixAja I can't get over how great the filtering is! Here's my comments:

### All Filters:
- [ ] [Should Have] When filter menu is open, filter dropdown component should be in active state instead of default (text color would be black, icon would be primary400)
- [ ] [Should Have] Text size for filter dropdowns should be updated to use font mixin fontCapsXxs (11px)
![Filtering-text updates](https://user-images.githubusercontent.com/63013360/133694719-15216fd7-88cc-439a-8694-655ecddb2576.png)

### Upload Date:
- [ ] [Nice to Have] Upload date is only supposed to have buckets for “Today”, “Yesterday”, and “Last 7 Days”
- [ ] [Should Have] - When error message appears for incorrect date format, can we have the inputs remain aligned to the top? 
![Filtering-error alignement](https://user-images.githubusercontent.com/63013360/133694812-9e50a420-e3ac-4b2e-9f51-ca80e68f0d60.png)
- [ ] [Must Have] - do not allow user to apply date range that is incorrectly formatted (disable button if format error is present)
![Filtering-error applied](https://user-images.githubusercontent.com/63013360/133694753-d7f540ee-f7a1-445f-8be2-0ab750cc325a.png)
- [ ] [Nice to Have] - I think we should only show the error message after use clicks out of the date input field (so they dont get an error before they finish typing)
- [ ] [Must Have, not a blocker] - when user makes a selection from the menu items (non-date range), the filter tag text should match. This ticket already exists [here](https://app.shortcut.com/genepi/story/161475/update-filter-tag-label-to-reflect-selection)

### Collection Date:
- [ ] Same as Upload Date, except that the menu item options are correct

### Lineage:
beautiful!

### Genome Recovery:
- [ ] [Must Have, not a blocker]  When I make a selection on Genome Recovery and then reopen the menu, the current selection should be using the “selected” state (text becomes semibold). Note: this may relate to [this ticket](https://app.shortcut.com/genepi/story/161482/remove-autofill-date-range-from-upload-date-collection-date-menu) for upload/collection date as well
![Filtering- active state](https://user-images.githubusercontent.com/63013360/133694866-9fc4e2ae-1647-44ee-87ee-2bb52422fc0b.png)

mmmmmmaya committed:
[ch160840] filter count styles (#674)

* filter panel toggle button styles

* types

* oops, extra style

mmmmmmaya committed:
[ch153503] phylo tree help link (#675)

* add icons

* add help link to tree page

* open link in new tab

mmmmmmaya committed:
tree type tooltips (#678)

* add header tooltip for tree type col in trees table

* add tooltip for tree type cells in tree table

* types

Commit:
add fontCapsXxxs (#680)

mmmmmmaya committed:
[sc161296] disable view download failed (#679)

* disable json download if tree not completed

* disable auspice link and modal when tree build not complete

mmmmmmaya committed:
Maya/161303 161299 tooltips (#682)

* move tree type col over

* tree table content style

* center creation date col

* underline tree type on hover

mmmmmmaya committed:
Maya/150460 status tags (#684)

* make column for tree name wider, as it must now also accomodate tag

* add status tags

* types

* don't expand sample headers though

vincent-czi committed:
Date Filtering fixes (#681)

* WIP: Ingest list of possible date options for menu

* WIP: Avoid state overlap between menu and field

* Handle async validation flicker, wrap up

* Fall in line with Prettier

* Backfill note on var name per Timmy

* Fix minor linter warning

* Integrate notes/suggestions from Timmy

* Show current selecting in Genome Recovery

Root cause here was due to the OPTIONS being set in the render of component.
But value checking to see what is selected is done on ===, so the option
objects never had equality because they were new objects each render.
By putting them outside component, they're always the same, and === works.

* date filter error message fix (#683)

* Update czifui to 0.0.55 (need API tweaks)

* Convert Genome Recovery filter to use ComplexFilter

* Styling tweaks for ComplexFilter

Co-authored-by: Phoenix Logan <phoenix@phoenixlogan.net>
